benchmark.py

```diff
@@ -344,6 +346,51 @@ def _run_benchmark(bazel_binary_path,
   return collected, (command, expressions, options)
 
 
+def handle_json_profiles_aggr(
+    bazel_commits, project_source, project_commits, runs, output_prefix,
+    output_path, data_directory):
+  """Aggregates the collected JSON profiles and writes the result to a CSV.
+
+   Args:
+    bazel_commits: the Bazel commits that bazel-bench ran on.
+    project_source:  a path/url to a local/remote repository of the project
+      on which benchmarking was performed.
+    project_commits: the commits of the project when benchmarking was done.
+    runs: the total number of runs.
+    output_prefix: the prefix to json profile filenames.
+      Often the bazel-bench-uid.
+    output_path: the path to the output csv file.
+    data_directory: the directory that stores output files.
+  """
+  output_dir = os.path.dirname(output_path)
+  if not os.path.exists(output_dir):
+    os.makedirs(output_dir)
+
+    with open(output_path, 'w') as f:
+      csv_writer = csv.writer(f)
+      csv_writer.writerow(
+          ['bazel_source', 'project_source', 'project_commit',
+           'cat', 'name', 'dur'])
+
+      for bazel_commit in bazel_commits:
+        for project_commit in project_commits:
```

Comment:
I am not rejecting the commit because of this but I would have chosen a different structure:
collect profile_paths in _run_benchmark, call directly to aggregate_data from there and then after all benchmarks are done write the collected aggregated data to file.

Reply:
I assume you mean maintaining a master list that contains all the aggregated data from each bazel_commit/project_commit pairs? `aggregate_data` is supposed to be oblivious of `bazel_commit` and `project_commit`, so the master list will have to patch those infos in, something like:
```
[{
  bazel_commit: 1,
  project_source: abc,
  project_commit: 2,
  cat: 'build phase markers',
  name: 'Analysis', dur: 10
},
...]
```
I didn't want to maintain another data structure so went with this approach instead.
Also the runs are pretty isolated from each other now, which is good. Having this master list would introduce a shared state across the runs?

Reply:
LGTM.